File: mongodb/documentation/source_code/task1/1e.js
db.Business.aggregate([
    {$lookup: {from: "Tip", localField: "business_id", foreignField: "business_id", as: "Tips"}},
    {$unwind: "$Tips"},
    {$match: {"Tips.date": {$gte: "2013-01-01", $lte: "2013-12-31"}}},
    {$group: {_id: "$name", amountOfTips: {$sum: 1}}},
    {$sort: {_id: 1}}
]);

File: mongodb/documentation/source_code/task1/1e.java
/*
private void executeTask1e() {
    AggregateIterable result = businessCollection.aggregate(
            asList(
                    lookup("Tip", "business_id", "business_id", "Tips"),
                    unwind("$Tips"),
                    match(and(
                            gte("Tips.date", "2013-01-01"),
                            lte("Tips.date", "2013-12-31")
                    )),
                    group("$name", sum("amountOfTips", 1)),
                    sort(ascending("_id"))
            ));

    for (Object o : result) {
        System.out.println(o);
    }
}
*/